drivers/counter/counter_mcux_tpm.c

```diff
@@ -282,7 +301,8 @@ static DEVICE_API(counter, mcux_tpm_driver_api) = {
 		.info = {							\
 			.max_top_value = TPM_MAX_COUNTER_VALUE(TPM(n)),		\
 			.freq = 0,						\
-			.channels = 1,						\
+			.channels = FSL_FEATURE_TPM_CHANNEL_COUNTn(		\
+					(TPM_Type *)DT_INST_REG_ADDR(n)),	\
```

Comment:
Seems like we can getting the number of channels in 2 ways. Can we use `MAX_CHANNELS` here as well?